tests/test_dispatch_client.py

```diff
@@ -81,24 +81,28 @@ async def test_list_dispatches() -> None:
         microgrid_id=1, value=[sampler.generate_dispatch() for _ in range(100)]
     )
 
-    async for dispatch in client.list(microgrid_id=1):
-        assert dispatch in client.dispatches(microgrid_id=1)
+    dispatches = client.list(microgrid_id=1)
+    async for page in dispatches:
+        for dispatch in page:
+            assert dispatch in client.dispatches(microgrid_id=1)
 
 
 async def test_list_create_dispatches() -> None:
     """Test listing dispatches."""
     sampler = DispatchGenerator()
     client = FakeClient()
 
-    assert await anext(client.list(microgrid_id=1), None) is None
+    page = await anext(client.list(microgrid_id=1))
+    assert len(page) == 0
 
     for i in range(100):
         sample = sampler.generate_dispatch()
         await client.create(**to_create_params(1, sample))
 
         dispatch = None
-        async for _dispatch in client.list(microgrid_id=1):
-            dispatch = _dispatch
+        async for page in client.list(microgrid_id=1):
+            for _dispatch in page:
+                dispatch = _dispatch
```

Comment:
Unrelated, but you should be able to do this too:

```suggestion
            for dispatch in page:
                pass
```

Variables in `for` in Python outlive their scope.

Reply:
strictly speaking, `for` variables don't introduce a new scope :P